test/topology_custom/test_tablets.py

```diff
@@ -301,3 +301,51 @@ async def test_read_of_pending_replica_during_migration(manager: ManagerClient,
 
     rows = await cql.run_async("SELECT pk from test.test")
     assert len(list(rows)) == 1
+
+
+@pytest.mark.asyncio
+@skip_mode('release', 'error injections are not supported in release mode')
+async def test_alter_dropped_tablets_keyspace(manager: ManagerClient) -> None:
+    config = {
+        'enable_tablets': 'true'
+    }
+
+    logger.info("starting a node (the leader)")
+    servers = [await manager.server_add(config=config)]
+
+    logger.info("starting a second node (the follower)")
+    servers += [await manager.server_add(config=config)]
+
+    await manager.get_cql().run_async("create keyspace ks with "
+                                      "replication = {'class': 'NetworkTopologyStrategy', 'replication_factor': 1} and "
+                                      "tablets = {'enabled': true}")
+    await manager.get_cql().run_async("create table ks.t (pk int primary key)")
+
+    logger.info(f"injecting wait-before-handling-keyspace-rf-change into the leader node {servers[0]}")
+    injection_handler = await inject_error_one_shot(manager.api, servers[0].ip_addr,
+                                                    'wait-before-handling-keyspace-rf-change')
+
+    async def alter_tablets_ks_without_waiting_to_complete():
+        res = await manager.get_cql().run_async("select data_center from system.local")
+        this_dc = res[0].data_center  # ALTER tablets KS needs to have the DC specified explicitly
+        manager.get_cql().run_async("alter keyspace ks "
+                                    f"with replication = {{'class': 'NetworkTopologyStrategy', '{this_dc}': 1}}")
```

Comment:
We should `await` for it and verify that it returns an error result at the end of the test.

You decided to spawn a task (`asyncio.create_task`) so the test case an progress concurrently while this is happening.

At the end of the test we can `await` the returned task:
```
task = asyncio.create_task(alter_tablets...)

... do the other stuff, drop keyspace, message handler ...

try:
    await task
except:
   ... verify that the returned exception is as expected (failed ALTER) ...
else:
   fail the test: we expected an exception, not successful response
```

Reply:
Rewritten, thanks.